nuxeo-features/nuxeo-annotation/nuxeo-annotation-core/src/main/java/org/nuxeo/ecm/annotation/AnnotationServiceImpl.java

```diff
@@ -56,11 +59,33 @@
 
     protected static final String GET_ANNOTATIONS_FOR_DOC_PAGEPROVIDER_NAME = "GET_ANNOTATIONS_FOR_DOCUMENT";
 
+    protected static final String ANNOTATIONS_PLACELESS_STORAGE_PROPERTY = "nuxeo.annotations.placeless.storage";
+
     @Override
     public Annotation createAnnotation(CoreSession session, Annotation annotation) {
-        // Create annotation as a placeless document
+
+        ConfigurationService configurationService = Framework.getService(ConfigurationService.class);
+        boolean annotationPlaceless = configurationService.isBooleanPropertyTrue(ANNOTATIONS_PLACELESS_STORAGE_PROPERTY);
+
         return CoreInstance.doPrivileged(session, s -> {
-            DocumentModel annotationModel = s.createDocumentModel(null, ANNOTATION_NAME, ANNOTATION_DOC_TYPE);
+            String path = null;
+            if (!annotationPlaceless) {
+                DocumentModel annotatedDoc = s.getDocument(new IdRef(annotation.getDocumentId()));
+                String parentPath = s.getRootDocument().getPathAsString();
+                if (annotatedDoc.getPath().segmentCount() > 1) {
```

Comment:
Can you add a comment stating that we try to put the document under the Domain ?